rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/ast/TypeInterestHandler.java

```diff
@@ -0,0 +1,35 @@
+package org.jboss.windup.rules.apps.java.scan.ast;
+
+import static org.joox.JOOX.$;
+
+import org.jboss.windup.config.exception.ConfigurationException;
+import org.jboss.windup.config.parser.ElementHandler;
+import org.jboss.windup.config.parser.NamespaceElementHandler;
+import org.jboss.windup.config.parser.ParserContext;
+import org.jboss.windup.config.parser.xml.RuleProviderHandler;
+import org.w3c.dom.Element;
+
+/**
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+@NamespaceElementHandler(elementName = TypeInterestHandler.ELEMENT_NAME, namespace = RuleProviderHandler.WINDUP_RULE_NAMESPACE)
+public class TypeInterestHandler implements ElementHandler<Void>
+{
+    public static final String ELEMENT_NAME = "type-interest";
+    private static final String PACKAGE_PREFIX = "package";
+    private static final String CLASSNAME = "class";
+    private static final String METHOD = "method";
+
+    @Override
+    public Void processElement(ParserContext handlerManager, Element element) throws ConfigurationException
```

Comment:
This is how we register interest in a particular package, class, or method. It avoids having to do extensive regex scans of everything.